### PR TITLE
jetbrains.clion: add patchelf invocations for new binaries

### PR DESCRIPTION
**Pull request by pmenke-de**
###### Motivation for this change
version 2021.1 of clion (introduced with 3839373) has introduced additional clang related binaries which need patching.
`clangd` and `clazy-standalone` also need the provided `libclazyPlugin.so` on the rpath.

I'm not really a specialist for patchelf or ld library loading. Especially i am unsure if the provided `libclazyPlugin.so.13git` needs to be / should be patchelf'd too. The application seems to "work for me" without that though.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 119905` at dd2c1dd5 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/1c55a890f287eb3e31566922abd05fc2">jetbrains.clion</a></li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement.

**Comment by Mic92** [comment]
The list grows longer. I think it would be time for autoPatchElfHook...

**Comment by Mic92** [comment]
I switched to autoPatchelfHook and it seems to work out bettter. However since I am not using this tool actively I have not tested everything yet. Could you have a look?

**Comment by pmenke-de** [comment]
Thanks for the change. It will test it later today.

File: pkgs/applications/editors/jetbrains/default.nix [hunk 1]
```diff
@@ -25,52 +28,23 @@ let
         platforms = platforms.linux;
       };
     }).overrideAttrs (attrs: {
+      nativeBuildInputs = (attrs.nativeBuildInputs or []) ++ optional (stdenv.isLinux) [
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/jetbrains/default.nix:31]:
```suggestion
      nativeBuildInputs = (attrs.nativeBuildInputs or []) ++ optionals (stdenv.isLinux) [
```

File: pkgs/applications/editors/jetbrains/default.nix [hunk 2]
```diff
@@ -25,52 +28,23 @@ let
...
+        autoPatchelfHook
+      ];
+      buildInputs = (attrs.buildInputs or []) ++ optional (stdenv.isLinux) [
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/jetbrains/default.nix:34]:
```suggestion
      buildInputs = (attrs.buildInputs or []) ++ optionals (stdenv.isLinux) [
```

**Comment by pmenke-de** [comment]
I tested the changes (up until d94d7c7) and everything still works for me.
Thank you for your reviews and suggestions :smile:

**Pull request merged**